src/clp_logging/handlers.py

```diff
@@ -792,3 +800,198 @@ def __init__(
         super().__init__(
             open(fpath, mode), enable_compression, timestamp_format, timezone, loglevel_timeout
         )
+
+
+class CLPS3Handler(CLPBaseHandler):
+    """
+    Log is written to stream in CLP IR encoding, and uploaded to s3_bucket
+
+    :param s3_bucket: S3 bucket to upload CLP encoded log messages to
+    """
+
+    def init(self, stream: IO[bytes]) -> None:
```

Comment:
what's the difference between this and the constructor? why do we want this as a public method?